contactsummary/js/src/contact-summary/controllers/key-dates.controller.js

```diff
@@ -46,47 +48,92 @@ define([
     }
 
     /**
-     * Fetch Contacts and Job Roles from Server
+     * Initialises subscribers
      */
-    function getContacts () {
-      resetKeyDates();
+    function initSubscribers () {
+      $log.debug('Subcsribers initialiazed!');
 
-      Contract.get()
-        .then(function (response) {
-          if (!_.isEmpty(response)) {
-            angular.forEach(response, function (contract) {
-              addContractDates(contract);
+      var events = [
+        'Contract::created',
+        'Contract::updated',
+        'JobRole::created',
+        'JobRole::updated',
+        'JobRole::deleted'
+      ];
 
-              if (contract.is_current === '1') {
-                vm.activeContracts++;
-              }
-            });
-          }
+      events.forEach(function (event) {
+        pubSub.subscribe(event, loadData);
+      });
 
-          return JobRole.get();
-        })
+      pubSub.subscribe('Contract::deleted', function (contract) {
+        Contract.removeContract(contract);
+        loadData();
+      });
+    }
+
+    /**
+     * Checks if a date is in the future
+     *
+     * @param  {String} date
+     * @return {Boolean}
+     */
+    function isDateNotInPast (date) {
+      return moment().diff(date, 'days') <= 0;
+    }
+
+    /**
+     * Loads contracts from the backend and sets their amount to the controller
+     *
+     * @return {Promise}
+     */
+    function loadContactsAmount () {
+      return Contract.get()
         .then(function (response) {
-          angular.forEach(response, function (role) {
-            var endDate = moment(role.end_date);
+          vm.activeContracts = _.values(response).filter(function (contract) {
+            addContractDates(contract);
```

Comment:
you are getting and setting at the same time?

Why not 

```js
var contracts = _.values(response);
vm.activeContracts = _.filter(contracts, 'is_current').length;

contracts.forEach(addContractDates);
```

Reply:
Also existing code, but let me check, good shout.